x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { SearchResponse } from 'elasticsearch';
+import { EndpointAppContext } from '../types';
+import { WhitelistRule } from '../../common/types';
+
+const whitelistIdx = 'whitelist-index';  // TODO: change this
+
+/**
+ * Registers the whitelist routes for the API
+ */
+export function registerWhitelistRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  router.get(
+    {
+      path: '/api/endpoint/whitelist',
+      validate: {},
+      options: { authRequired: true },
+    },
+    handleWhitelistGet
+  );
+
+  const INVALID_UUID = "`alert_id` Must be a UUID.'";
+  router.post(
+    {
+      path: '/api/endpoint/whitelist',
+      validate: {
+        body: schema.object({
+          comment: schema.maybe(schema.string()), // Optional comment explaining reason for whitelist
+          alert_id: schema.string({
+            minLength: 36, // https://tools.ietf.org/html/rfc4122#section-3
+            maxLength: 36,
+            validate(value) {
+              // Must be a UUID
+              if (!validateUUID(value)) {
+                return INVALID_UUID;
+              }
+            },
+          }),
+          file_path: schema.maybe(schema.string()),
+          signer: schema.maybe(schema.string()),
+          sha256: schema.maybe(schema.string()),
```

Comment:
Would this satisfy the `macroHash` whitelist requirement as well or do we need another request param for it?

Reply:
not sure, that wasn't mentioned in the issue that I based this off of: https://github.com/elastic/endpoint-dev/issues/6532

Reply:
It's mentioned here https://github.com/elastic/endpoint-dev/issues/6532#issuecomment-586462136. We should look into it.

Reply:
It's mentioned here https://github.com/elastic/endpoint-dev/issues/6532#issuecomment-586462136. We should look into it.